packages/property-grid/src/components/PropertyGridUiItemsProvider.tsx

```diff
@@ -1,48 +1,53 @@
 /*---------------------------------------------------------------------------------------------
-* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
-* See LICENSE.md in the project root for license terms and full copyright notice.
-*--------------------------------------------------------------------------------------------*/
-
-import { connect } from "react-redux";
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
 
 import { I18N } from "@bentley/imodeljs-i18n";
 import {
   AbstractWidgetProps,
   StagePanelLocation,
   StagePanelSection,
+  StageUsage,
   UiItemsProvider,
 } from "@bentley/ui-abstract";
-import { FrameworkState } from "@bentley/ui-framework";
-import { PropertyGrid, PropertyGridProps } from "./PropertyGrid";
+import { UiFramework } from "@bentley/ui-framework";
+import { PropertyGridProps } from "../property-grid-react";
 import * as React from "react";
-
-// Map framework state to property grid props
-const mapStateToProps = (state: FrameworkState): PropertyGridProps => {
-  return {
-    iModelConnection: state.sessionState.iModelConnection,
-    projectId: state.sessionState.iModelConnection.iModelToken.contextId,
-  };
-};
-
-// tslint:disable-next-line: variable-name
-const ConnectedPropertyGrid = connect(mapStateToProps)(PropertyGrid);
+import { PresentationPropertyGridWidget } from "./PresentationPropertyGridWidget2";
 
 /** Provides the property grid widget to zone 9 */
 export class PropertyGridUiItemsProvider implements UiItemsProvider {
   public readonly id = "PropertyGridUiitemsProvider";
   public static i18n: I18N;
 
+  private _props?: PropertyGridProps;
+
+  constructor(props?: PropertyGridProps) {
+    this._props = props;
+  }
+
   public provideWidgets(
     _stageId: string,
-    _stageUsage: string,
+    stageUsage: string,
     location: StagePanelLocation,
-    _section: StagePanelSection | undefined,
+    _section: StagePanelSection | undefined
   ): ReadonlyArray<AbstractWidgetProps> {
     const widgets: AbstractWidgetProps[] = [];
-    if (location === StagePanelLocation.Right) {
+    if (
+      stageUsage === StageUsage.General &&
+      location === StagePanelLocation.Right
+    ) {
+      // this._props
+
       widgets.push({
         id: "propertyGrid",
-        getWidgetContent: () => <ConnectedPropertyGrid />,
+        label: "Properties",
+        getWidgetContent: () => (
+          <PresentationPropertyGridWidget
+            projectId={UiFramework.getIModelConnection()?.contextId ?? ""}
```

Comment:
are we ignoring all the props passed into the provider? 
As mentioned before do the same thing you did in the tree-widget; if user passes in an imodel connection use that, if not then get the imodel connection like this and pass it in.